pkg/kubectl/cmd/util/helpers.go

```diff
@@ -461,3 +465,47 @@ func UpdateObject(info *resource.Info, updateFn func(runtime.Object) error) (run
 
 	return info.Object, nil
 }
+
+// GetThirdPartyGroupVersions loads dynamic resource versions
+func GetThirdPartyGroupVersions(discovery client.DiscoveryInterface) ([]unversioned.GroupVersion, []unversioned.GroupVersionKind, error) {
+	result := []unversioned.GroupVersion{}
+	gvks := []unversioned.GroupVersionKind{}
+
+	groupList, err := discovery.ServerGroups()
+	if err != nil {
+		// On forbidden or not found, just return empty lists.
+		if errors.IsForbidden(err) || errors.IsNotFound(err) {
+			return result, gvks, nil
+		}
+
+		return nil, nil, err
+	}
+
+	for ix := range groupList.Groups {
+		group := &groupList.Groups[ix]
+		for jx := range group.Versions {
+			gv, err2 := unversioned.ParseGroupVersion(group.Versions[jx].GroupVersion)
+			if err2 != nil {
+				return nil, nil, err
+			}
+			// Skip GroupVersionKinds that have been statically registered.
+			if registered.IsRegisteredVersion(gv) {
+				continue
+			}
+			result = append(result, gv)
+
+			resourceList, err := discovery.ServerResourcesForGroupVersion(group.Versions[jx].GroupVersion)
+			if err != nil {
+				return nil, nil, err
+			}
+			for kx := range resourceList.APIResources {
+				gvks = append(gvks, gv.WithKind(resourceList.APIResources[kx].Kind))
+			}
+		}
+	}
+	return result, gvks, nil
+}
+
+func AddDiscoveryFlags(cmd *cobra.Command) {
+	cmd.Flags().Bool("enable-dynamic-discovery", true, "If true, perform dynamic discovery of new APIs via calls to the API server.")
```

Comment:
I'd still like someone else from @kubernetes/kubectl  to vet the name.  Maybe @jlowdermilk?  This is pretty widespread.